irr/src/CIrrDeviceSDL.cpp

```diff
@@ -700,12 +700,11 @@ bool CIrrDeviceSDL::run()
 {
 	os::Timer::tick();
 
-	SEvent irrevent;
 	SDL_Event SDL_event;
 
 	while (!Close && wrap_PollEvent(&SDL_event)) {
+		SEvent irrevent{};
```

Comment:
A pitfall here is that aggregate-initialization only initializes the first member of an `union`.
what we have to do here is `memset(&irrevent, 0, sizeof(irrevent));` (and then you can also move the declaration back)